website/docs/r/data_forwarding_rule.html.markdown

```diff
@@ -9,25 +9,61 @@ description: |-
 Provider to manage [Sumologic Data Forwarding Rule](https://help.sumologic.com/docs/manage/data-forwarding/amazon-s3-bucket/#forward-datato-s3)
 
 ## Example Usage
+
+For Partitions
 ```hcl
+resource "sumologic_partition" "test_partition" {
+  name               = "testing_rule_partitions"
+  routing_expression = "_sourcecategory=abc/Terraform"
+  is_compliant       = false
+  retention_period   = 30
+  analytics_tier     = "flex"
+}
+
 resource "sumologic_data_forwarding_rule" "example_data_forwarding_rule" {
-    index_id = "00000000024C6155"
-    destination_id = "00000000000732AA"
-    enabled = "true"
-    file_format = "test/{index}/{day}/{hour}/{minute}"
-    payload_schema = "builtInFields"
-    format = "json"
+  index_id = sumologic_partition.test_partition.id
+  destination_id = "00000000000732AA"
+  enabled = true
+  file_format = "test/{index}/{day}/{hour}/{minute}"
+  payload_schema = "builtInFields"
+  format = "json"
+}
+```
+For Scheduled Views
+```hcl
+resource "sumologic_scheduled_view" "failed_connections" {
+  index_name = "failed_connections"
+  query = "_sourceCategory=fire | count"
+  start_time = "2024-09-01T00:00:00Z"
+  retention_period = 1
+  lifecycle {
+    prevent_destroy = true
+    ignore_changes = [index_id]
+  }
+}
+
+resource "sumologic_data_forwarding_rule" "test_rule_sv" {
+  index_id       = sumologic_scheduled_view.failed_connections.index_id
+  destination_id = sumologic_data_forwarding_destination.test_destination.id
+  enabled        = false
+  file_format    = "test/{index}"
+  payload_schema = "raw"
+  format         = "text"
 }
 ```
 ## Argument reference
 
 The following arguments are supported:
 
-- `index_id` - (Required) The _id_ of the Partition or Scheduled View the rule applies to.
+- `index_id` - (Required) The *id* of the Partition or *index_id* of the Scheduled View the rule applies to.
 - `destination_id` - (Required) The data forwarding destination id.
-- `enabled` - (Required) True when the data forwarding rule is enabled.
+- `enabled` - (Optional) True when the data forwarding rule is enabled. Will be treated as _false_ if left blank.
```

Comment:
Should we add possible values for `file_format` here to aid customers?

Reply:
Not needed, we already have a descriptive documentation ... and that is added on top of this page - [see here](https://github.com/SumoLogic/terraform-provider-sumologic/pull/691/files#diff-92753b9c76a682404721409b68e2d58d2edadcd8e4ae69ef0ffe01cb66ec65ebR9)

Reply:
>Not needed

I disagree, we should cover the possible values here. Can't rely on an example to cover to do that.

Reply:
Shall I point to the [original documentation here](https://help.sumologic.com/docs/manage/data-forwarding/amazon-s3-bucket/#forward-datato-s3) (point 6) in that case? 

Possible values for `file_format` are a combination of few placeholders, and adding all of them here will not look good in the TF docs imho.

Reply:
Updated the docs in the [latest](https://github.com/SumoLogic/terraform-provider-sumologic/pull/691/commits/651ae5560867ff1dc5318edd273fe2ae42588b39) commit.